### PR TITLE
WIP: Allowing annotations and content to be refreshed

### PR DESCRIPTION
**Pull request by jpbida**
Wanted to be able to submit annotations and then refresh the output with new text. 
Issues: 
   - calling Recogito.init in render would recursive wrap the text-content 
   - the event handlers createAnnotation & updateAnnotation would be rebound each rerender 
   - on rerender clear annotations and refresh event handlers

### CODE REVIEW

**Comment by jpbida** [comment]
Thanks for creating the recogito widget! 
Any thoughts on 
 - moving the Recogito.init call into the factory function 
 - passing the updateAnnotations & createAnnotations as options in the widget 
 - using r._environment from the Recogito library

File: inst/htmlwidgets/recogito.js [hunk 1]
```diff
@@ -44,10 +25,35 @@ HTMLWidgets.widget({
           }
           r.setMode(annotationMode);
         });
+        function updateAnnotation(){
+           Shiny.setInputValue(r._environment.inputId, JSON.stringify(r.getAnnotations()));
```
Comment (jpbida) [inline inst/htmlwidgets/recogito.js:29]: Was thinking of passing updateAnnotation & createAnnotations as options. Using r._environment to pass the inputId. Is there a better way?

**Comment by jwijffels** [comment]
Thanks, appreciate the input. Can we make this into 2 pull requests, 

1. one for upgrading recogito to >= 1.7, while still making sure the current setup works and 
2. another one on the improvements on the updating of the existing annotations.

**Comment by jpbida** [comment]
@jwijffels Created a pull request for just the recogito 1.7.1 update and added some automated testing.

File: inst/htmlwidgets/recogito.js (hunk 1 above)
Comment (jwijffels) [reply inst/htmlwidgets/recogito.js:29]: I should probably return r (as shown at https://www.htmlwidgets.org/develop_intro.html) such that you can interact with it

**Pull request closed without merging**